testng-eclipse-plugin/src/main/org/testng/eclipse/launch/TestNGLaunchConfigurationDelegate.java

```diff
@@ -134,18 +134,7 @@ protected VMRunnerConfiguration launchTypes(final ILaunchConfiguration configura
     }
 
     // Program & VM args
-    StringBuilder vmArgs = new StringBuilder(ConfigurationHelper.getJvmArgs(configuration))
-        .append(" ")
-        .append(TestNGLaunchConfigurationConstants.VM_ENABLEASSERTION_OPTION); // $NON-NLS-1$
-    addDebugProperties(vmArgs, configuration);

```

Comment:
move all JVM args computation into ConfigurationHelper.getJvmArgs() so that we can have consistent result from different run/debug entry points